bundles/org.openhab.binding.monopriceaudio/src/main/java/org/openhab/binding/monopriceaudio/internal/handler/MonopriceAudioHandler.java

```diff
@@ -0,0 +1,706 @@
+import org.openhab.binding.monopriceaudio.internal.communication.MonopriceAudioMessageEvent;
+import org.openhab.binding.monopriceaudio.internal.communication.MonopriceAudioMessageEventListener;
+import org.openhab.binding.monopriceaudio.internal.communication.MonopriceAudioSerialConnector;
+import org.openhab.binding.monopriceaudio.internal.communication.MonopriceAudioZone;
+import org.openhab.binding.monopriceaudio.internal.configuration.MonopriceAudioThingConfiguration;
+import org.openhab.binding.monopriceaudio.internal.dto.MonopriceAudioZoneDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link MonopriceAudioHandler} is responsible for handling commands, which are sent to one of the channels.
+ *
+ * Based on the Rotel binding by Laurent Garnier
+ *
+ * @author Michael Lobstein - Initial contribution
+ */
+@NonNullByDefault
+public class MonopriceAudioHandler extends BaseThingHandler implements MonopriceAudioMessageEventListener {
+    private static final long RECON_POLLING_INTERVAL_SEC = 60;
+    private static final long INITIAL_POLLING_DELAY_SEC = 5;
+    private static final Pattern PATTERN = Pattern
+            .compile("^(\\d{2})(\\d{2})(\\d{2})(\\d{2})(\\d{2})(\\d{2})(\\d{2})(\\d{2})(\\d{2})(\\d{2})(\\d{2})");
+
+    private static final String ZONE = "ZONE";
+    private static final String ALL = "all";
+    private static final String CHANNEL_DELIMIT = "#";
+    private static final String ON_STR = "01";
+    private static final String OFF_STR = "00";
+
+    private static final int ONE = 1;
+    private static final int MAX_ZONES = 18;
+    private static final int MAX_SRC = 6;
+    private static final int MIN_VOLUME = 0;
+    private static final int MAX_VOLUME = 38;
+    private static final int MIN_TONE = -7;
+    private static final int MAX_TONE = 7;
+    private static final int MIN_BALANCE = -10;
+    private static final int MAX_BALANCE = 10;
+    private static final int BALANCE_OFFSET = 10;
+    private static final int TONE_OFFSET = 7;
+
+    private final Logger logger = LoggerFactory.getLogger(MonopriceAudioHandler.class);
+    private final MonopriceAudioStateDescriptionOptionProvider stateDescriptionProvider;
+    private final SerialPortManager serialPortManager;
+
+    private @Nullable ScheduledFuture<?> reconnectJob;
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    private MonopriceAudioConnector connector = new MonopriceAudioDefaultConnector();
+
+    private HashSet<String> ignoreZones = new HashSet<>();
+    private long lastPollingUpdate = System.currentTimeMillis();
+    private long pollingInterval = 0;
+    private int numZones = 0;
+    private int allVolume = 1;
+    private int initialAllVolume = 0;
+    private Object sequenceLock = new Object();
+
+    // build a Map with a MonopriceAudioZoneDTO for each zoneId
+    private final Map<String, MonopriceAudioZoneDTO> ZONE_DATA_MAP = MonopriceAudioZone.VALID_ZONE_IDS.stream()
```

Comment:
```suggestion
    private final Map<String, MonopriceAudioZoneDTO> zoneDataMap = MonopriceAudioZone.VALID_ZONE_IDS.stream()
```

Also move this field above the non-final fields.

Reply:
done